components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2878,50 +2884,18 @@ OAuthProblemException buildOAuthProblemException(AuthenticationResult authentica
 
     /**
      * Store sessionID using the redirect URl.
-     *
-     * @param redirectURL
+     * @param oAuthMessage
      * @param sessionState
+     * @param redirectURL
      */
-    private void storeSidClaim(String redirectURL, OIDCSessionState sessionState, OAuth2Parameters oAuth2Parameters) {
+    private void storeSidClaim(OAuthMessage oAuthMessage, OIDCSessionState sessionState, String redirectURL) {
 
-        String idToken;
-        String code;
         if (redirectURL.contains(ID_TOKEN)) {
-
-            try {
-                if (isFormPostResponseMode(oAuth2Parameters, redirectURL)) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("form_post response mode is enabled and redirectURL is in valid JSON format for " +
-                                "clientID : " + oAuth2Parameters.getClientId());
-                    }
-                    JSONObject jsonData = new JSONObject(redirectURL);
-                    idToken = (String) jsonData.get(ID_TOKEN);
-                } else {
-                    idToken = getIdTokenFromRedirectURL(redirectURL);
-                }
-                if (!idToken.isEmpty()) {
-                    addSidToSessionStateFromIdToken(idToken, sessionState);
-                }
-            } catch (URISyntaxException e) {
-                log.error("Error while getting ID token from redirectURL ", e);
-            }
+            String oidcSessionState = (String) oAuthMessage.getProperty(OIDC_SESSION_ID);
```

Comment:
Have we tested this with implicit flow?

Reply:
yes. tested implicit flow with and without id_token encryption.